lib/index.ts

```diff
@@ -28,7 +28,17 @@ export class FilePath {
 
   /** Get: the current path */
   public get path(): string {
-    return `${this.folders.join(sep)}${sep}${this.file ? this.file : ''}`;
+    return join(this.folders.join(sep), this.file ? this.file : sep);
+  }
+
+  /** Get: the current directory */
+  public get dir(): string {
+    return join(this.folders.join(sep));
+  }
+
+  /** Set: change the current directory */
+  public set dir(dir: string) {
+    this.path = join(dir, this.file ? this.file : '');
```

Comment:
I'd say set the folders attribute directly, as in;

```js
this.folders = dir.split(sep);
```

Adding the file to the join in `dir` doesn't actually change anything.

I think from an API standpoint it makes sense to keep the folders array the same as what was input (so a user using the index to manipulate the folders will have expected results), but normalize the output.

Here is what I would do (link to branch); 

https://github.com/autovance/filepath/blob/658755351db7322f5795dd8f0134757bf9d06ed0/lib/index.ts#L42